homeassistant/components/climate/ambiclimate.py

```diff
@@ -0,0 +1,271 @@
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_HUMIDITY, DOMAIN, SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_ON_OFF, STATE_HEAT)
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (ATTR_NAME, ATTR_TEMPERATURE,
+                                 STATE_OFF, TEMP_CELSIUS)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+REQUIREMENTS = ['ambiclimate==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+AMBICLIMATE_DATA = 'ambiclimate'
+ATTR_VALUE = 'value'
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONFIGURATOR_DESCRIPTION = 'To link your Ambiclimate account, ' \
+                           'click the link, login, and authorize:'
+CONFIGURATOR_LINK_NAME = 'Link Ambiclimate account'
+CONFIGURATOR_SUBMIT_CAPTION = 'I authorized successfully'
+DEFAULT_NAME = 'Ambiclimate'
+SERVICE_COMFORT_FEEDBACK = 'send_comfort_feedback'
+SERVICE_COMFORT_MODE = 'set_comfort_mode'
+SERVICE_TEMPERATURE_MODE = 'set_temperature_mode'
+STORAGE_KEY = 'ambiclimate_auth'
+STORAGE_VERSION = 1
+
+AUTH_CALLBACK_NAME = 'api:ambiclimate'
+AUTH_CALLBACK_PATH = '/api/ambiclimate'
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_ON_OFF)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENT_ID): cv.string,
+    vol.Required(CONF_CLIENT_SECRET): cv.string,
+})
+
+SEND_COMFORT_FEEDBACK_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_VALUE): cv.string,
+})
+
+SET_COMFORT_MODE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+})
+
+SET_TEMPERATURE_MODE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_VALUE): cv.string,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Ambiclimate heater."""
+    import ambiclimate
+    callback_url = '{}{}'.format(hass.config.api.base_url, AUTH_CALLBACK_PATH)
+    oauth = ambiclimate.AmbiclimateOAuth(config.get(CONF_CLIENT_ID),
+                                         config.get(CONF_CLIENT_SECRET),
+                                         callback_url,
+                                         async_get_clientsession(hass))
+
+    store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+    token_info = await store.async_load()
+
+    if not token_info:
+        _LOGGER.info("no token; requesting authorization")
+        hass.http.register_view(AmbiclimateAuthCallbackView(config,
+                                                            async_add_entities,
+                                                            oauth,
+                                                            store))
+
+        def _call_request_config():
+            configurator = hass.components.configurator
```

Comment:
Per the comment about moving to a config flow, this configurator logic becomes unnecessary as you can now perform this sequence during setup of the integration via config flow steps.